06_functions/main.go

```diff
@@ -74,41 +73,134 @@ Note: functions can be defined in any order in the file.
 
 func main() {
 	sayHello("John")
+	fmt.Println(diff(1, 9))
+	fmt.Println(diff(9, 1))
+	fmt.Println(diff(4, -10))
+	printInput("안녕", 3)
+	printInput("go", 7)
+	printInput("ça va", 4)
+	fmt.Println(concatString("CatDog", 5))
+	fmt.Println(concatString("Rice", 5))
+	fmt.Println(concatString("김치", 6))
+	printToInt(4)
+	printToInt(9)
+	printToInt(6)
+	fmt.Println(returnStrings("Anna", "Hanna"))
+	length("Gopher")
+	length("Go rocks")
+	lengthAdvanced("A")
+	lengthAdvanced("")
+	lengthAdvanced("Luna")
+	fmt.Println(doingMath(3, 4))
+}
 
-	// For each exercise, call the functions a couple of times with different
-	// values and print the results.
+// For each exercise, call the functions a couple of times with different
+// values and print the results.
 
-	// TODO: Write a function that takes two integers as parameters and returns
-	// the difference of the two integers.
+// TODO: Write a function that takes two integers as parameters and returns
+// the difference of the two integers.
+func diff(a int, b int) int {
+	var diff int = b - a
+	return diff
+}
 
-	// TODO: Write a function that takes a string and an integer as parameters.
-	// The function should print the string as many times as the integer.
+// TODO: Write a function that takes a string and an integer as parameters.
+// The function should print the string as many times as the integer.
+func printInput(str string, times int) {
+	var result = ""
+	var count = 1
+	for {
+		if count > times {
+			break
+		}
+		fmt.Println(str)
+		count++
+	}
+	fmt.Println(result)
+}
 
-	// TODO: Write a function that takes a string and an integer as parameters.
-	// The function should concatenate (i.e. join) the string with itself as
-	// many times as the integer and return the result.
-	// Call the function a couple of times with different values and print the
-	// results.
+// TODO: Write a function that takes a string and an integer as parameters.
+// The function should concatenate (i.e. join) the string with itself as
+// many times as the integer and return the result.
+// Call the function a couple of times with different values and print the
+// results.
+func concatString(str2 string, times2 int) string {
+	var count2 = 1
+	// var result2 = ""
+	for {
+		if count2 > times2 {
+			break
+		}
+		str2 += str2
+		count2++
+	}
+	return str2
+}
```

Comment:
There's a way to write that without an additional counter.
Can you figure out how?

Reply:
Try writing the same function without using `count2`.

Hint:
```go
func concatString(str2 string, times2 int) string {
	for {
		if ??? > 0 {
			break
		}
		str2 += str2

		???
	}
	return str2
}
```

Reply:
True, then maybe this: ```go
func concatString(str2 string, times2 int) string {
        var result string = ""
	for {
		if ??? > 0 {
			break
		}
		result += str2

		???
	}
	return result
}
```

Reply:
I did think of declaring another string too, that seems the only way I can think of at least.

Reply:
How about now? If you want `> 0` in the logic, I would give it another go.

Reply:
I think the code looks good now, as long as it prints the correct result :)